plugins/org.eclipse.sirius.tests.junit/src/org/eclipse/sirius/tests/unit/diagram/layout/SimpleELKLayoutTest.java

```diff
@@ -1658,12 +1836,122 @@ protected void testArrangeAllResult_ForPackageResetOrigin(String diagramName, bo
+
+        // Check that the migration of the session resource is needed.
+        Version loadedVersion = checkRepresentationFileMigrationStatus(URI.createPlatformPluginURI("/" + TEMPORARY_PROJECT_NAME + "/" + REPRESENTATIONS_RESOURCE_NAME, true), true);
+        assertTrue("The migration must be required on test data.", migrationVersion.compareTo(loadedVersion) > 0); //$NON-NLS-1$
+    }
+
+    private void checkRoutingStyle(AbstractDiagramEdgeEditPart edgeEditPart, String message, String edgeName, int expectedRoutingStyle) {
+        String fullMessage = MessageFormat.format(message, edgeName);
+        View notationView = edgeEditPart.getNotationView();
+        DDiagramElement dde = edgeEditPart.resolveDiagramElement();
+        assertTrue("The diagram element of the AbstractDiagramEdgeEditPart must be a DEdge.", dde instanceof DEdge);
+        DEdgeQuery dEdgeQuery = new DEdgeQuery((DEdge) dde);
+        Routing routing = dEdgeQuery.getRouting();
+        assertEquals(fullMessage, expectedRoutingStyle, routing.getValue());
+        // Check that custom features list contains the routingStyle (if necessary)
+        Option<EdgeMapping> optionalEdgeMapping = new IEdgeMappingQuery(((DEdge) dde).getActualMapping()).getEdgeMapping();
+        if (optionalEdgeMapping.some()) {
+            EdgeRouting originalEdgeRouting = optionalEdgeMapping.get().getStyle().getRoutingStyle();
+            if (EdgeRouting.get(expectedRoutingStyle).equals(originalEdgeRouting)) {
+                // The routing is the same, so the edge should not appear as customized.
+                assertTrue("This edge \"" + edgeName + "\" must not appear as customized.",
+                        (!(new DDiagramElementQuery(dde).isCustomized() || new ViewQueryWithoutFontNameCheck(notationView).isCustomized())));
+            } else {
+                assertTrue("The custom features list of the style of DEdge \"" + edgeName + "\" must contain the routingStyle.",
+                        dde.getStyle().getCustomFeatures().contains(DiagramPackage.Literals.DEDGE__ROUTING_STYLE.getName()));
+            }
+        }
+    }
+
     protected void openDiagram(String diagramName) {
         diagram = (DDiagram) getRepresentationsByName(diagramName).toArray()[0];
         editorPart = (IDiagramWorkbenchPart) DialectUIManager.INSTANCE.openEditor(session, diagram, new NullProgressMonitor());
         TestsUtil.synchronizationWithUIThread();
+        checkStringValueStyle();
+    }
+
+    protected void checkStringValueStyle() {
+        Diagram gmfDiag = getGmfDiagram(diagram);
+        if (gmfDiag != null) {
+            TreeIterator<EObject> childIterator = gmfDiag.eAllContents();
+            while (childIterator.hasNext()) {
+                EObject eObject = childIterator.next();
+                if (eObject instanceof Edge) {
+                    Edge edge = (Edge) eObject;
```

Comment:
You can use Java 17 instanceof:
```suggestion
                if (eObject instanceof Edge edge) {
```

Reply:
It will be fixed in next version.